### ISSUE TITLE
Resources not loading from module

### ISSUE THREAD

**Issue by carltonlee1** [hashicorp/terraform#23030]
### Terraform Version

```
0.11.14
```

### Terraform Configuration Files

tfstate
```
{
    "version": 3,
    "serial": 1,
    "lineage": "c9261ae7-b6c8-82d3-0a82-c8f937549f7c",
    "backend": {
        "type": "s3",
        "config": {
            "bucket": "****terraform-states",
            "dynamodb_table": "****terraform-locks",
            "encrypt": true,
            "key": "core/<account_id>/terraform.tfstate",
            "region": "us-east-1",
            "role_arn": "arn:aws:iam::<account_id>:role/*****Admin"
        },
        "hash": 10574190286517913165
    },
    "modules": [
        {
            "path": [
                "root"
            ],
            "outputs": {},
            "resources": {},
            "depends_on": []
        }
    ]
}

```
backend.tf
```
terraform {
  backend "s3" {
    role_arn = "arn:aws:iam::<account_id>:role/*****Admin"
    region   = "us-east-1"
    bucket   = "***terraform-states"
    key      = "core/<account_id>/terraform.tfstate"
    encrypt  = true

    dynamodb_table = "****terraform-locks"
  }
}

```
provider
```
provider "aws" {
  assume_role {
    role_arn = "arn:aws:iam::${var.account_id}:role/${var.assume_role}"
  }
}

data "aws_caller_identity" "current" {}
```
### Expected Behavior

New policy and IAM resources should have been created within the individual account.
### Actual Behavior

No new resources are noted within the plan or apply. They do, however, show up in other accounts.

We use modules to create cross-account roles in ~90+ accounts within AWS. This has worked up until recently (in one particular account - folder structure below). We had to restore the remote state file to an older version to get TF to work. We're not sure what happened, but our more recent state files didn't have data in them. After the restore, everything seemed to work for local accounts. However, cross-account roles created within the modules folder no longer worked in this particular account. They work in all the other accounts, however.

Each folder (account) has its own state file.

Long story short:

After restoring a backup state file, modules are no longer working on one particular folder.

### Additional Context

Folder structure:
```
Modules/
--main.tf
--variables.tf
--policies.tf
--roles.tf
--outputs.tf

Account1/
--main.tf
--variables.tf
--outputs.tf
--localPolicies.tf


Account1/
--main.tf
--variables.tf
--outputs.tf
--localPolicies.tf
```

**Comment by carltonlee1** [comment]
Also, looks like serial on the remote version is WAAAAY off... Not sure how to fix.

```    "version": 3,
    "terraform_version": "0.11.14",
    "serial": 238,
```

**Comment by teamterraform** [comment]
Hi @carltonlee1! Sorry things aren't working as expected.

It looks like none of the configuration snippets you shared include any `resource` or `module` blocks, and so Terraform planning to do nothing would be the expected behavior here. Do you have additional configuration files that you didn't include here?

**Comment by carltonlee1** [comment]
The folder structure above is relevant here. This works in all accounts except one. The folder structure is identical for all accounts. This is the new role and policies that will be applied to all accounts.

modules/roles.tf

```
module "X-Role-SecOps" {
  source               = "../x-role"
  name                 = "CloudCoreSecOps"
  assume_role_policy   = "${local.assume_role_policy}"
  max_session_duration = "${local.max_session_duration}"
  permissions_boundary = "${aws_iam_policy.L3Permissions.arn}"
}

resource "aws_iam_role_policy_attachment" "X-Role-SecOps" {
  count = 1
  role  = "${module.X-Role-SecOps.name}"

  policy_arn = "${element(list(
    aws_iam_policy.SecOpsRole.arn
    ), count.index)}"
}
```
modules/policies.tf
```
data "aws_iam_policy_document" "SecOps" {
  statement {
    sid       = "SecOpsGuardDutyAcess"
    effect    = "Allow"
    resources = ["*"]

    actions = [
      "guardduty:Get*",
      "guardduty:List*",
      "guardduty:Describe*",
      "guardduty:UpdateIPSet",
      "guardduty:UpdateThreatIntelSet",
      "guardduty:UpdateFilter",
      "guardduty:DeleteIPSet",
      "guardduty:DeleteThreatIntelSet",
      "guardduty:DeleteFilter",
    ]
  }
}

resource "aws_iam_policy" "SecOps" {
  name        = "CloudCoreSecOps"
  description = "[Terraform] CloudCore policy for SecOps teams to manager GuardDuty and Alerting."
  policy      = "${data.aws_iam_policy_document.SecOps.json}"
}
```

**Comment by carltonlee1** [comment]
Account1 Plan log example with expected output (account2 doesn't do this):
```
  + module.CloudCore.module.X-Role-SecOps.aws_iam_role.main
      id:                    <computed>
      arn:                   <computed>
      assume_role_policy:    "{\n  \"Version\": \"2012-10-17\",\n  \"Statement\": [\n    {\n      \"Sid\": \"\",\n      \"Effect\": \"Allow\",\n      \"Action\": \"sts:AssumeRoleWithSAML\",\n      \"Principal\": {\n        \"Federated\": \"arn:aws:iam::<account_num>:saml-provider/aig-prod-ADFS\"\n      },\n      \"Condition\": {\n        \"StringEquals\": {\n          \"SAML:aud\": \"https://signin.aws.amazon.com/saml\"\n        }\n      }\n    }\n  ]\n}"
      create_date:           <computed>
      description:           "[Terraform] cross-account access role"
      force_detach_policies: "false"
      max_session_duration:  "43200"
      name:                  "CloudCoreSecOps"
      path:                  "/"
      permissions_boundary:  "arn:aws:iam::<account_num>:policy/CloudCoreL3Permissions"
      tags.%:                "2"
      tags.ManagedBy:        "CloudServices/Terraform"
      tags.Name:             "CloudCoreSecOps"
      unique_id:             <computed>```

**Comment by danieldreier** [comment]
I want to apologize for the slow response time on this issue, and also let you know that I am bulk-closing all issues exclusively reported against Terraform 0.11.x, including this issue, because we are no longer investigating issues reported against Terraform 0.11.x. In most cases, when we try to reproduce issues reported against 0.11, we either can't reproduce them anymore, or the reporter has moved on, so we believe we can better support the Terraform user community by prioritizing more recent issues.

Terraform 0.12 has been available since May of 2019, and there are really significant benefits to adopting it. I know that migrating from 0.11 to versions past 0.12 can require a bit of effort, but it really is worth it, and the upgrade path is pretty well understood in the community by now. 0.14 is available and stable, and we are quickly approaching an 0.15 release.

We have made a significant effort in the last year to stay on top of bug reports; we have triaged almost all new bug reports within 1-2 weeks for 6+ months now. If you are still experiencing this problem, please submit a new bug report with a reproduction case that works on 0.14.x, link this old issue for context, and we will triage it.
